### ISSUE TITLE
Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached

### ISSUE THREAD

**Issue by vulpes2** [flatpak/xdg-desktop-portal#986]
Note: this is not the common problem where applications that use portals take forever to start, this is about xdg-desktop-portal _itself_ taking an unusual amount of time to start.

OS: Fedora 38 (beta)
Desktop Environment: Plasma 5.27.3
xdg-desktop-portal: 1.16.0

After upgrading my system, I noticed a lot of programs take almost forever to start, and it ended up being caused by xdg-desktop-portal not running as it should be. Later I found out xdg-desktop-portal now takes over a minute to start, which is longer than the default timeout (60s). 

This can be partially mitigated by extending the `TimeoutSec` setting for the systemd unit , but I would still have to wait over a minute after logging in to be able to launch most of my applications. I'm not quite sure how to debug this any further, any help or information would be greatly appreciated.

```
14:34:54 pc systemd[2271]: Stopped xdg-desktop-portal.service - Portal service.
14:34:54 pc systemd[2271]: Starting xdg-desktop-portal.service - Portal service...
14:35:44 pc xdg-desktop-por[64232]: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
14:36:09 pc xdg-desktop-por[64232]: Failed to create wallpaper proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
14:36:09 pc xdg-desktop-por[64232]: No skeleton to export
14:36:09 pc systemd[2271]: Started xdg-desktop-portal.service - Portal service.
```

```
$ /usr/libexec/xdg-desktop-portal --replace --verbose
XDP: load portals from /usr/share/xdg-desktop-portal/portals
XDP: loading /usr/share/xdg-desktop-portal/portals/kde.portal
XDP: portal implementation for KDE
XDP: portal implementation supports org.freedesktop.impl.portal.Access
XDP: portal implementation supports org.freedesktop.impl.portal.Account
XDP: portal implementation supports org.freedesktop.impl.portal.AppChooser
XDP: portal implementation supports org.freedesktop.impl.portal.Background
XDP: portal implementation supports org.freedesktop.impl.portal.Email
XDP: portal implementation supports org.freedesktop.impl.portal.FileChooser
XDP: portal implementation supports org.freedesktop.impl.portal.Inhibit
XDP: portal implementation supports org.freedesktop.impl.portal.Notification
XDP: portal implementation supports org.freedesktop.impl.portal.Print
XDP: portal implementation supports org.freedesktop.impl.portal.ScreenCast
XDP: portal implementation supports org.freedesktop.impl.portal.Screenshot
XDP: portal implementation supports org.freedesktop.impl.portal.RemoteDesktop
XDP: portal implementation supports org.freedesktop.impl.portal.Settings
XDP: portal implementation supports org.freedesktop.impl.portal.DynamicLauncher
XDP: portal implementation supports org.freedesktop.impl.portal.GlobalShortcuts
XDP: loading /usr/share/xdg-desktop-portal/portals/gtk.portal
XDP: portal implementation for gnome
XDP: portal implementation supports org.freedesktop.impl.portal.FileChooser
XDP: portal implementation supports org.freedesktop.impl.portal.AppChooser
XDP: portal implementation supports org.freedesktop.impl.portal.Print
XDP: portal implementation supports org.freedesktop.impl.portal.Notification
XDP: portal implementation supports org.freedesktop.impl.portal.Inhibit
XDP: portal implementation supports org.freedesktop.impl.portal.Access
XDP: portal implementation supports org.freedesktop.impl.portal.Account
XDP: portal implementation supports org.freedesktop.impl.portal.Email
XDP: portal implementation supports org.freedesktop.impl.portal.DynamicLauncher
XDP: portal implementation supports org.freedesktop.impl.portal.Settings
XDP: loading /usr/share/xdg-desktop-portal/portals/gnome-keyring.portal
XDP: portal implementation for gnome
XDP: portal implementation supports org.freedesktop.impl.portal.Secret
XDP: loading /usr/share/xdg-desktop-portal/portals/gnome.portal
XDP: portal implementation for gnome
XDP: portal implementation supports org.freedesktop.impl.portal.Account
XDP: portal implementation supports org.freedesktop.impl.portal.AppChooser
XDP: portal implementation supports org.freedesktop.impl.portal.Screenshot
XDP: portal implementation supports org.freedesktop.impl.portal.ScreenCast
XDP: portal implementation supports org.freedesktop.impl.portal.RemoteDesktop
XDP: portal implementation supports org.freedesktop.impl.portal.Lockdown
XDP: portal implementation supports org.freedesktop.impl.portal.Background
XDP: portal implementation supports org.freedesktop.impl.portal.Settings
XDP: portal implementation supports org.freedesktop.impl.portal.Wallpaper
XDP: portal implementation supports org.freedesktop.impl.portal.FileChooser
XDP: portal implementation supports org.freedesktop.impl.portal.Print
XDP: portal implementation supports org.freedesktop.impl.portal.DynamicLauncher
XDP: Falling back to gnome.portal for org.freedesktop.impl.portal.Lockdown
XDP: providing portal org.freedesktop.portal.MemoryMonitor
XDP: providing portal org.freedesktop.portal.PowerProfileMonitor
XDP: providing portal org.freedesktop.portal.NetworkMonitor
XDP: providing portal org.freedesktop.portal.ProxyResolver
XDP: providing portal org.freedesktop.portal.Trash
XDP: providing portal org.freedesktop.portal.GameMode
XDP: providing portal org.freedesktop.portal.Realtime
XDP: Using kde.portal for org.freedesktop.impl.portal.Settings
XDP: Using gnome.portal for org.freedesktop.impl.portal.Settings
XDP: Using gtk.portal for org.freedesktop.impl.portal.Settings

(/usr/libexec/xdg-desktop-portal:65292): xdg-desktop-portal-WARNING **: 14:46:20.075: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
XDP: providing portal org.freedesktop.portal.Settings
XDP: Using kde.portal for org.freedesktop.impl.portal.FileChooser in KDE
XDP: providing portal org.freedesktop.portal.FileChooser
XDP: Using kde.portal for org.freedesktop.impl.portal.AppChooser in KDE
XDP: providing portal org.freedesktop.portal.OpenURI
XDP: Using kde.portal for org.freedesktop.impl.portal.Print in KDE
XDP: providing portal org.freedesktop.portal.Print
XDP: Using kde.portal for org.freedesktop.impl.portal.Notification in KDE
XDP: providing portal org.freedesktop.portal.Notification
XDP: Using kde.portal for org.freedesktop.impl.portal.Inhibit in KDE
XDP: providing portal org.freedesktop.portal.Inhibit
XDP: Using kde.portal for org.freedesktop.impl.portal.Access in KDE
XDP: Using kde.portal for org.freedesktop.impl.portal.Screenshot in KDE
XDP: providing portal org.freedesktop.portal.Screenshot
XDP: Using kde.portal for org.freedesktop.impl.portal.Background in KDE
XDP: providing portal org.freedesktop.portal.Device
XDP: providing portal org.freedesktop.portal.Location
XDP: providing portal org.freedesktop.portal.Camera

(/usr/libexec/xdg-desktop-portal:65292): xdg-desktop-portal-WARNING **: 14:46:20.111: Failed to create background monitor: Failed to own background monitor D-Bus name

(/usr/libexec/xdg-desktop-portal:65292): xdg-desktop-portal-WARNING **: 14:46:20.111: No skeleton to export
XDP: Falling back to gnome.portal for org.freedesktop.impl.portal.Wallpaper

(/usr/libexec/xdg-desktop-portal:65292): xdg-desktop-portal-WARNING **: 14:46:45.137: Failed to create wallpaper proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached

(/usr/libexec/xdg-desktop-portal:65292): xdg-desktop-portal-WARNING **: 14:46:45.137: No skeleton to export
XDP: Using kde.portal for org.freedesktop.impl.portal.Account in KDE
XDP: using org.freedesktop.impl.portal.Account at org.freedesktop.impl.portal.desktop.kde

XDP: providing portal org.freedesktop.portal.Account
XDP: Using kde.portal for org.freedesktop.impl.portal.Email in KDE
XDP: providing portal org.freedesktop.portal.Email
XDP: Falling back to gnome-keyring.portal for org.freedesktop.impl.portal.Secret
XDP: providing portal org.freedesktop.portal.Secret
XDP: Using kde.portal for org.freedesktop.impl.portal.GlobalShortcuts in KDE
XDP: providing portal org.freedesktop.portal.GlobalShortcuts
XDP: Using kde.portal for org.freedesktop.impl.portal.DynamicLauncher in KDE
XDP: providing portal org.freedesktop.portal.DynamicLauncher
XDP: Using kde.portal for org.freedesktop.impl.portal.ScreenCast in KDE
XDP: providing portal org.freedesktop.portal.ScreenCast
XDP: Using kde.portal for org.freedesktop.impl.portal.RemoteDesktop in KDE
XDP: providing portal org.freedesktop.portal.RemoteDesktop
XDP: org.freedesktop.portal.Desktop acquired
```

**Comment by vulpes2** [comment]
Uninstalling `xdg-desktop-portal-gnome` fixed the timeout, now `xdg-desktop-portal` can start almost instantly as expected. Interestingly this package is also installed on Fedora 37's KDE spin and it's enabled by default, however it doesn't seem to be causing any issues on F37, only on F38.

**Comment by jadahl** [comment]
This should be addressed with https://github.com/flatpak/xdg-desktop-portal/pull/985 and https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/78.

**Comment by q234rty** [comment]
Still happens for me on x-d-p 1.16.0 + cherry-picked https://github.com/flatpak/xdg-desktop-portal/commit/2a219279997c2124c8a639b0d009c9946ec97b40 + x-d-p-gnome 44.1

In particular, 
```
Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
```
still happens.
I _think_ the reason is that #985 does not change the behavior of `find_all_portal_implementations` so x-d-p still times out waiting for x-d-p-gnome while creating the settings portal.

**Comment by jadahl** [comment]
> I think the reason is that https://github.com/flatpak/xdg-desktop-portal/pull/985 does not change the behavior of find_all_portal_implementations so x-d-p still times out waiting for x-d-p-gnome while creating the settings portal.

That's odd, I could reproduce the problem before (using KDE), and with the updated xdp-gnome and xdp it didn't time out anymore. What distribution are you using?

**Comment by q234rty** [comment]
> What distribution are you using?

I'm using Arch Linux with an admittedly funky setup (greetd/sway/etc.)

I have tested a bit more and found that:

1. Under dbus-broker, the issue does seem to be resolved after #985 and the settings portal does try to start x-d-p-gnome but it fails immediately w/o hitting a timeout: `Failed to ReadAll() from Settings implementation: GDBus.Error:org.freedesktop.DBus.Error.NameHasNoOwner: Could not activate remote peer: startup job failed.` and it falls back to the x-d-p-gtk as it should.

2. Under dbus-daemon, even after #985 and removing `org.freedesktop.impl.portal.Settings` from `gnome.portal`, x-d-p seems to hit the the timeout when creating the dummy proxy:
`XDP: Failed to create dummy fallback proxy on org.freedesktop.impl.portal.desktop.gnome for org.freedesktop.impl.portal.Print: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached`

**Comment by x-yuri** [comment]
Yeah, I can confirm this on Arch Linux:

```
May 16 15:48:30 yuri2 systemd[1100]: Starting Portal service...
May 16 15:48:30 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gnome' unit='xdg-desktop-portal-gnome.service' requested by ':1.72' (uid=1000 pid=15331 comm="/usr/lib/xdg-desktop-portal")
May 16 15:48:30 yuri2 systemd[1100]: Current graphical user session is inactive.
May 16 15:48:30 yuri2 systemd[1100]: Dependency failed for Portal service (GNOME implementation).
May 16 15:48:30 yuri2 systemd[1100]: xdg-desktop-portal-gnome.service: Job xdg-desktop-portal-gnome.service/start failed with result 'dependency'.
May 16 15:49:20 yuri2 xdg-desktop-por[15331]: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
May 16 15:50:00 yuri2 systemd[1100]: xdg-desktop-portal.service: start operation timed out. Terminating.
May 16 15:50:00 yuri2 systemd[1100]: xdg-desktop-portal.service: Failed with result 'timeout'.
May 16 15:50:00 yuri2 systemd[1100]: Failed to start Portal service.
```

Removing [`xdg-desktop-portal-gnome`][a] seems to solve the issue :) At least `xdg-desktop-portal` restarts. And the applications that need `xdg-desktop-portal` seem to work (at least they start, not hang).

The strange thing is that for a while (after upgrade) it seemed to work, but at some point it stopped.

Not sure if I need `xdg-desktop-portal-gnome`, since I'm using [`awesome`][b] windows manager. But sometimes I run GNOME apps.

[a]: https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-gnome/
[b]: https://awesomewm.org/

**Comment by q234rty** [comment]
@x-yuri : Try switching to dbus-broker by installing it and running `sudo systemctl enable --global dbus-broker.service` and `sudo systemctl enable dbus-broker.service`. This has resolved the issue for me even with x-d-p-gnome installed.

**Comment by x-yuri** [comment]
@q234rty Actually, another option I've found is downgrading `xdg-desktop-portal-gnome` to 43.1. But thanks for the suggestion, maybe I'll need it at some point.

It seems to have stopped working since `xdg-desktop-portal-gnome-44.0`.

`xdg-desktop-portal-1.16.0` + `xdg-desktop-portal-gnome-44.0`:

```
$ journalctl -b USER_UNIT=xdg-desktop-portal.service + USER_UNIT=xdg-desktop-portal-gnome.service + USER_UNIT=dbus.service + _SYSTEMD_USER_UNIT=xdg-desktop-portal.service + _SYSTEMD_USER_UNIT=xdg-desktop-portal-gnome.service + _SYSTEMD_USER_UNIT=dbus.service
May 16 16:32:04 yuri2 systemd[1101]: Starting Portal service...
May 16 16:32:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.freedesktop.portal.Documents' unit='xdg-document-portal.service' requested by ':1.14' (uid=1000 pid=1275 comm="/usr/lib/xdg-desktop-portal")
May 16 16:32:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.freedesktop.impl.portal.PermissionStore' unit='xdg-permission-store.service' requested by ':1.15' (uid=1000 pid=1299 comm="/usr/lib/xdg-document-portal")
May 16 16:32:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
May 16 16:32:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Successfully activated service 'org.freedesktop.portal.Documents'
May 16 16:32:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gnome' unit='xdg-desktop-portal-gnome.service' requested by ':1.14' (uid=1000 pid=1275 comm="/usr/lib/xdg-desktop-portal")
May 16 16:32:04 yuri2 systemd[1101]: Starting Portal service (GNOME implementation)...
May 16 16:32:04 yuri2 xdg-desktop-portal-gnome[1328]: Failed to initialize display server connection: Unsupported or missing session type ''
May 16 16:32:04 yuri2 systemd[1101]: xdg-desktop-portal-gnome.service: Main process exited, code=exited, status=1/FAILURE
May 16 16:32:04 yuri2 systemd[1101]: xdg-desktop-portal-gnome.service: Failed with result 'exit-code'.
May 16 16:32:04 yuri2 systemd[1101]: Failed to start Portal service (GNOME implementation).
May 16 16:32:12 yuri2 systemd[1008]: Stopping D-Bus User Message Bus...
May 16 16:32:12 yuri2 systemd[1008]: Stopped D-Bus User Message Bus.
May 16 16:32:29 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gtk' unit='xdg-desktop-portal-gtk.service' requested by ':1.14' (uid=1000 pid=1275 comm="/usr/lib/xdg-desktop-portal")
May 16 16:32:29 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.a11y.Bus' unit='at-spi-dbus-bus.service' requested by ':1.50' (uid=1000 pid=2237 comm="/usr/lib/xdg-desktop-portal-gtk")
May 16 16:32:29 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Successfully activated service 'org.a11y.Bus'
May 16 16:32:29 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Successfully activated service 'org.freedesktop.impl.portal.desktop.gtk'
May 16 16:32:54 yuri2 xdg-desktop-por[1275]: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
May 16 16:33:34 yuri2 systemd[1101]: xdg-desktop-portal.service: start operation timed out. Terminating.
May 16 16:33:34 yuri2 systemd[1101]: xdg-desktop-portal.service: Failed with result 'timeout'.
May 16 16:33:34 yuri2 systemd[1101]: Failed to start Portal service.
May 16 16:34:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Failed to activate service 'org.freedesktop.portal.Desktop': timed out (service_start_timeout=120000ms)
May 16 16:34:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.28' (uid=1000 pid=1246 comm="telegram-desktop")
May 16 16:34:04 yuri2 dbus-daemon[1120]: [session uid=1000 pid=1120] Failed to activate service 'org.freedesktop.impl.portal.desktop.gnome': timed out (service_start_timeout=120000ms)
```

Here of note is supposedly:

```
May 16 16:32:04 yuri2 xdg-desktop-portal-gnome[1328]: Failed to initialize display server connection: Unsupported or missing session type ''
```

`xdg-desktop-portal-1.16.0` + `xdg-desktop-portal-gnome-44.1`:

```
$ journalctl -b USER_UNIT=xdg-desktop-portal.service + USER_UNIT=xdg-desktop-portal-gnome.service + USER_UNIT=dbus.service + _SYSTEMD_USER_UNIT=xdg-desktop-portal.service + _SYSTEMD_USER_UNIT=xdg-desktop-portal-gnome.service + _SYSTEMD_USER_UNIT=dbus.service
May 16 17:40:37 yuri2 systemd[1086]: Starting Portal service...
May 16 17:40:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.freedesktop.portal.Documents' unit='xdg-document-portal.service' requested by ':1.14' (uid=1000 pid=1259 comm="/usr/lib/xdg-desktop-portal")
May 16 17:40:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.freedesktop.impl.portal.PermissionStore' unit='xdg-permission-store.service' requested by ':1.15' (uid=1000 pid=1289 comm="/usr/lib/xdg-document-portal")
May 16 17:40:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
May 16 17:40:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Successfully activated service 'org.freedesktop.portal.Documents'
May 16 17:40:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gnome' unit='xdg-desktop-portal-gnome.service' requested by ':1.14' (uid=1000 pid=1259 comm="/usr/lib/xdg-desktop-portal")
May 16 17:40:37 yuri2 systemd[1086]: Dependency failed for Portal service (GNOME implementation).
May 16 17:40:37 yuri2 systemd[1086]: xdg-desktop-portal-gnome.service: Job xdg-desktop-portal-gnome.service/start failed with result 'dependency'.
May 16 17:40:46 yuri2 systemd[994]: Stopping D-Bus User Message Bus...
May 16 17:40:46 yuri2 systemd[994]: Stopped D-Bus User Message Bus.
May 16 17:41:02 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gtk' unit='xdg-desktop-portal-gtk.service' requested by ':1.14' (uid=1000 pid=1259 comm="/usr/lib/xdg-desktop-portal")
May 16 17:41:02 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.a11y.Bus' unit='at-spi-dbus-bus.service' requested by ':1.50' (uid=1000 pid=2242 comm="/usr/lib/xdg-desktop-portal-gtk")
May 16 17:41:02 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Successfully activated service 'org.a11y.Bus'
May 16 17:41:02 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Successfully activated service 'org.freedesktop.impl.portal.desktop.gtk'
May 16 17:41:28 yuri2 xdg-desktop-por[1259]: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.gnome: Timeout was reached
May 16 17:42:07 yuri2 systemd[1086]: xdg-desktop-portal.service: start operation timed out. Terminating.
May 16 17:42:07 yuri2 systemd[1086]: xdg-desktop-portal.service: Failed with result 'timeout'.
May 16 17:42:07 yuri2 systemd[1086]: Failed to start Portal service.
May 16 17:42:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Failed to activate service 'org.freedesktop.portal.Desktop': timed out (service_start_timeout=120000ms)
May 16 17:42:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.29' (uid=1000 pid=1231 comm="telegram-desktop")
May 16 17:42:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Failed to activate service 'org.freedesktop.impl.portal.desktop.gnome': timed out (service_start_timeout=120000ms)
```

What's wrong here is not particularly clear, inactive session?

```
May 16 17:42:37 yuri2 systemd[1086]: Starting Portal service...
May 16 17:42:37 yuri2 dbus-daemon[1103]: [session uid=1000 pid=1103] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gnome' unit='xdg-desktop-portal-gnome.service' requested by ':1.54' (uid=1000 pid=2954 comm="/usr/lib/xdg-desktop-portal")
May 16 17:42:37 yuri2 systemd[1086]: Current graphical user session is inactive.
May 16 17:42:37 yuri2 systemd[1086]: Dependency failed for Portal service (GNOME implementation).
May 16 17:42:37 yuri2 systemd[1086]: xdg-desktop-portal-gnome.service: Job xdg-desktop-portal-gnome.service/start failed with result 'dependency'.
```

`xdg-desktop-portal-1.16.0` + `xdg-desktop-portal-gnome-43.1`: seems to work.

**Comment by akana** [comment]
> @x-yuri : Try switching to dbus-broker by installing it and running `sudo systemctl enable --global dbus-broker.service` and `sudo systemctl enable dbus-broker.service`. This has resolved the issue for me even with x-d-p-gnome installed.

this solved my problem. thanks you

**Comment by smcv** [comment]
Switching to dbus-broker is a workaround here, for systemd not reporting services that fail to start in a way that dbus-daemon understands.

**Comment by RafaelPalomar** [comment]
> @x-yuri : Try switching to dbus-broker by installing it and running `sudo systemctl enable --global dbus-broker.service` and `sudo systemctl enable dbus-broker.service`. This has resolved the issue for me even with x-d-p-gnome installed.

This solved the problem for me on ArchLinux and makes Firefox not taking 20 secs to start.

**Comment by jamesob** [comment]
Once I installed and enabled `dbus-broker` per instructions above, and uninstalled `xdg-desktop-portal-gnome`, the issue resolved.

**Comment by GeorgesStavracas** [comment]
Thanks everyone who worked on it. Considering this fixed. Closing.

**Comment by luispabon** [comment]
Sorry @GeorgesStavracas, could you please let us know how is it fixed? Was it a change to xdg-desktop-portal-gnome or something else?

**Comment by blurgyy** [comment]
> Sorry @GeorgesStavracas, could you please let us know how is it fixed? Was it a change to xdg-desktop-portal-gnome or something else?

@luispabon According to this thread, it seems the solution is to switch the system dbus implementation to `dbus-broker`.

**Comment by aleb** [comment]
In case anyone is wondering, dbus-broker is a replacement of the dbus reference implementation. This seems like a hack. Why is there a difference between dbus-broker and the reference implementation―which one is the "correct" behavior and which has to be fixed for real?

**Comment by x-yuri** [comment]
> According to this thread, it seems the solution is to switch the system dbus implementation to dbus-broker.

@blurgyy According to this thread switching to `dbus-broker` is the workaround that is [no longer needed][a].

> Why is there a difference in behavior between `dbus-broker` and the reference implementation―which one is the "correct" behavior?

@aleb You can find the answer if you just [read the issue][b].

[a]: https://github.com/flatpak/xdg-desktop-portal/issues/986#issuecomment-1745095729
[b]: https://github.com/flatpak/xdg-desktop-portal/issues/986#issuecomment-1551524524

**Comment by blurgyy** [comment]
Hi @x-yuri, this thread does not seem to mention in which version this issue got fixed.  In my case with `xdg-desktop-portal 1.16.0`, `xdg-desktop-portal-wlr 0.7.0`, `xdg-desktop-portal-gtk 1.14.1`, the issue only disappears after using dbus-broker.  I also find that switching to `dbus-broker` is more like a workaround, but I'm happy with it so far.  Perhaps you can elaborate on how to achieve the exact fix so that further readers can avoid this confusion ;)

**Comment by x-yuri** [comment]
Well, I said "according to this thread." I don't have `xdg-desktop-portal-gnome` installed on my current machine. But [here][a] they say:

> This should be fixed for systems using dbus-broker in 44.1 and dbus-daemon in 44.2.

I guess the way it was fixed can be found [here][b].

I see that you have `xdg-desktop-portal-wlr`, `xdg-desktop-portal-gtk`. Not `xdg-desktop-portal-gnome`. My understanding is that the solultion applies only to the GNOME package. Actually the issue is about GNOME.

Maybe @GeorgesStavracas has anything to add?

[a]: https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/86#note_1802681
[b]: https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/78

**Comment by x-yuri** [comment]
Arch Linux switched [from `dbus-daemon` to `dbus-broker`][a].

[a]: https://archlinux.org/news/making-dbus-broker-our-default-d-bus-daemon/

**Comment by AgarwalPragy** [comment]
Issue still exists with Ubuntu 24.04

> Uninstalling `xdg-desktop-portal-gnome` fixed the timeout, now `xdg-desktop-portal` can start almost instantly as expected. Interestingly this package is also installed on Fedora 37's KDE spin and it's enabled by default, however it doesn't seem to be causing any issues on F37, only on F38.
> 
> https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/74

uninstalling fixes it
